File: CalcNF.cpp
#include <stdio.h>
#include <stdlib.h>

int main() {
    float media1, mediaf;
    system("title CalcNF - Por Diemousine");

    while(1>0) {
    system("cls");
    printf("Obs: use . e nao ,\n\nDigite sua media: ");
    scanf("%f",&media1);

    if (media1>=7) { 
       printf("Voce nao precisa fazer prova final.\n\n");
       } else {
              mediaf = ((1-((media1*0.7)/5))*5)/0.3;
              if(mediaf>10) {
                            printf("Nao tem santo no mundo que faca voce passar na final!\n\n");
              } else {
                   printf("Voce precisa tirar %f para passar na Prova Final.\n\n",mediaf);
                   }
              }
    system("Pause");
}
    return 0;
}

File: CalcNF.java
import java.util.Scanner;
public class CalcNF {

	public static void main(String[] args) {
		Scanner meuScanner = new Scanner(System.in);
		float mediaAtual;
		double mediaFinal;
		System.out.println("Digite o valor da sua média atual");
		mediaAtual = meuScanner.nextFloat();
		if(mediaAtual>=7){
			System.out.println("Você não precisa fazer a prova final."); 
			} else {
				mediaFinal = ((1-((mediaAtual*0.7)/5))*5)/0.3;
				if(mediaFinal>10){
					System.out.println("Não tem santo no mundo que faça você passar.");
				} else {
						System.out.println("Você precisa tirar "+mediaFinal+" para passar.");
			}
		}
	}
}